### ISSUE TITLE
ALIAS for @endcode fails with warning: reached end of file while inside a 'code' block! The command that should end the block seems to be missing!

### ISSUE THREAD

**Issue by scott-zhong** [doxygen/doxygen#10676]
ALIAS for `@endcode` fails with `warning: reached end of file while inside a 'code' block! The command that should end the block seems to be missing!` on trunk (6822b2f151227e144f699990929afe6dbf667fee*).  Does not fail with 1.10.0.

```
/**
 * \starttest
 * Something
 * \endtest
 *
 */
class Test
{
};
```

with configurations

```
# Difference with default Doxyfile 1.11.0 (6822b2f151227e144f699990929afe6dbf667fee*)
ALIASES                = "starttest=\par testing ^^\code" \
                         endtest=\endcode
```

produces warning

```
test.h:10: warning: reached end of file while inside a 'code' block! The command that should end the block seems to be missing!
```

testcase:

[testcase.zip](https://github.com/doxygen/doxygen/files/14351025/testcase.zip)

**Comment by doxygen** [comment]
@albert-github The best I can think of is to evaluate aliases also inside `\code`, but only push their expansion back into the parse stream when it contains an `\endcode`. The same can be done for the other verbatim blocks. This keeps the "verbatim" property of these commands, without missing an end command hidden in an alias.

**Comment by doxygen** [comment]
@scott-zhong Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official release becomes available.

**Comment by albert-github** [comment]
@doxygen I think the verbatim block ` ``` ` (and `~~~`) are not taken into consideration.

Example: [example.tar.gz](https://github.com/doxygen/doxygen/files/14395860/example.tar.gz)

**Comment by doxygen** [comment]
> @doxygen I think the verbatim block `` ``` `` (and `~~~`) are not taken into consideration.
> 
> Example: [example.tar.gz](https://github.com/doxygen/doxygen/files/14395860/example.tar.gz)

They are taken into account, but this seems to be problem with `^^` which causes the start of the verbatim block to appear after a `\ilinebr` whereas later processing expects it at the start of a real newline. This was already an issue with earlier versions as well. I'll look into a fix.

**Comment by doxygen** [comment]
@albert-github Should hopefully be fixed now.

**Comment by albert-github** [comment]
@doxygen Looks good, thanks.

**Comment by scott-zhong** [comment]
@doxygen Yes, the commit addresses the problem. Thank you!!!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.11.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
